src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -401,6 +418,102 @@ private CloudResourceManager cloudResourceManager() throws IOException, GeneralS
             .build();
     }
 
+    /**
+     * Create a client to speak to the appengine admin api
+     */
+    private Appengine appengine() throws GeneralSecurityException, IOException {
```

Comment:
Not sure what the code conventions in this file are, but it looks to me that the new code is a separate set of operations from `GoogleProjectService`. Once you have the app engine object, the operations are really on that object, not on this service object. And once those operations are complete, the object is no longer needed. So I would put this code in a constructor, and put `enableFirestore()` and `blockUntilAppengineOperationComplete()` as methods on the new class.

Reply:
I think this is following the current convention of the file...looks like we already have similar patterns of execute and block on operation for different operation objects (which are just different enough to make it a pain to reuse)